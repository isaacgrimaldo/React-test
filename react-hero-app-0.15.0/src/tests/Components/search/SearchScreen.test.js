import React from 'react';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { MemoryRouter, Route } from 'react-router';

import { SearchScreen } from '../../../components/search/SearchScreen';
import { history } from '../../features/testContex'

describe('Pruebas del componente SearchScreen.js', () => {
    
    it('Debe de mostrar correctamente el componete', () => {
        
        const wrapper = mount(
             <MemoryRouter initialEntries ={['/search']}>
                  <Route exact path='/search' component = { SearchScreen } />
             </MemoryRouter>
        );

        expect(wrapper.find('h1').text().trim()).toBe('Search Screen');
        expect(wrapper).toMatchSnapshot();
    });
       
    it('Debe de cambiar el valor del input y este tiene que ser el mismo que hay en la URL', () =>{
        const wrapper = mount(
            <MemoryRouter initialEntries ={['/search?q=Hulk']}>
                 <Route exact path='/search' component = { SearchScreen } />
            </MemoryRouter>
       );
       
       expect(wrapper.find('input').prop('value')).toBe('Hulk');
       expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar un error cuando el heroe no se encuentra', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries ={['/search?q=noHERO']}>
                 <Route exact path='/search' component = { SearchScreen } />
            </MemoryRouter>
       );

       expect(wrapper.find('.alert-danger').text().trim()).toBe('There is no a hero with noHERO');
       expect(wrapper).toMatchSnapshot();
    });

    test('Dede de hacer los cambio correctamente en el input y debe llamar al push cuando se hace Submit', () => {
        
        const wrapper = mount(
            <MemoryRouter initialEntries ={['/search']}>
                 <Route exact path='/search' component = { () => <SearchScreen history = { history  }/> } />
            </MemoryRouter>
       );
       
       wrapper.find('input').simulate('change',{
           target:{
               name:'searchText',
               value:'Hulk'
           }
       });
       
       wrapper.find('form').prop('onSubmit') ({
        preventDefault(){}
     }) 
       
       const { push } = history;
       expect( push ).toHaveBeenCalledWith(`?q=Hulk`)       
    });
});