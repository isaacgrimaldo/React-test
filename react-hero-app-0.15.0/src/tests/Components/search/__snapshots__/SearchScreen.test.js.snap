// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente SearchScreen.js Debe de cambiar el valor del input y este tiene que ser el mismo que hay en la URL 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
         Search Form 
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value="Hulk"
        />
        <button
          className="btn m-1 btn-block btn-outline-primary"
          type="submit"
        >
          Search...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
         Results 
      </h4>
      <hr />
      <div
        className="card ms-3 animate__animated animate__fadeIn"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Hulk"
              className="card-img"
              src="./assets/heroes/marvel-hulk.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Hulk
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Bruce Banner
                 
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                   
                  The Incredible Hulk #1
                   
                </small>
              </p>
              <a
                href="/hero/marvel-hulk"
                onClick={[Function]}
              >
                MÃ¡s...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Pruebas del componente SearchScreen.js Debe de mostrar correctamente el componete 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
         Search Form 
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value=""
        />
        <button
          className="btn m-1 btn-block btn-outline-primary"
          type="submit"
        >
          Search...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
         Results 
      </h4>
      <hr />
      <div
        className="alert alert-info"
      >
        Search a hero
      </div>
    </div>
  </div>
</div>
`;

exports[`Pruebas del componente SearchScreen.js debe de mostrar un error cuando el heroe no se encuentra 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
         Search Form 
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value="noHERO"
        />
        <button
          className="btn m-1 btn-block btn-outline-primary"
          type="submit"
        >
          Search...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
         Results 
      </h4>
      <hr />
      <div
        className="alert alert-danger"
      >
        There is no a hero with 
        noHERO
      </div>
    </div>
  </div>
</div>
`;
